libs/sdk-react-native/example/App.js

```diff
@@ -145,6 +145,7 @@ const App = () => {
                 addLine("backupStatus", JSON.stringify(await backupStatus()))
             } catch (e) {
                 addLine("error", e.toString())
+                console.log(`Error: ${JSON.stringify(e)}`)
```

Comment:
How would you handle this error here in real life?
I guess you don't really know whether it's a SendPaymentFailed error or something else here?
What would the logic to do something based on the error type look like?

Reply:
It's a bit more difficult in RN as you only have the option to pass a string code and message. The error e would look something like:
```
{
    code: "SendPaymentFailed.AlreadyPaid",
    message: "Invoice already paid",
    ...
}
```
You would probably map the error code to translation key to get the correct language for an error message, or do some feedback/logic based on the code key.

Reply:
It's not possible to add extra fields into the promise reject from native android/ios to typescript 😒 

Here's the interface:
```
reject(code, message, e)
```

And the result:
```
{
  "code": "ServiceError.Generic",
  "message": "No LSP found for id 03cea51f-b654-4fb0-8e82-eca137f236a0",
  "domain": "BreezSDK.SdkError",
  "userInfo": {},
  "nativeStackIOS": [
    "0   BreezSDKExample                     0x000000011003faf0 RCTJSErrorFromCodeMessageAndNSError + 112",
    "1   BreezSDKExample                     0x000000010ffd1b1a __41-[RCTModuleMethod processMethodSignature]_block_invoke_2.129 + 170",
    "2   BreezSDKExample                     0x000000010fa99b17 $sSo8NSStringCSgACSo7NSErrorCSgIeyByyy_SSSgAGs5Error_pSgIegggg_TR + 359",
    "3   BreezSDKExample                     0x000000010faabde6 $s8BreezSDK07RNBreezB0C9rejectErr3err0D0ys5Error_p_ySSSg_AHsAG_pSgtctF + 4710",
    "4   BreezSDKExample                     0x000000010faa5dee $s8BreezSDK07RNBreezB0C7lspInfo_6rejectyyypSgc_ySSSg_AGs5Error_pSgtctF + 638",
    "5   BreezSDKExample                     0x000000010faa6032 $s8BreezSDK07RNBreezB0C7lspInfo_6rejectyyypSgc_ySSSg_AGs5Error_pSgtctFTo + 162",
    "6   CoreFoundation                      0x00007ff80042f2fc __invoking___ + 140",
    "7   CoreFoundation                      0x00007ff80042c529 -[NSInvocation invoke] + 305",
    "8   CoreFoundation                      0x00007ff80042c7bc -[NSInvocation invokeWithTarget:] + 70",
    "9   BreezSDKExample                     0x000000010ffd3536 -[RCTModuleMethod invokeWithBridge:module:arguments:] + 2534",
    "10  BreezSDKExample                     0x000000010ffd79da _ZN8facebook5reactL11invokeInnerEP9RCTBridgeP13RCTModuleDatajRKN5folly7dynamicEiN12_GLOBAL__N_117SchedulingContextE + 1402",
    "11  BreezSDKExample                     0x000000010ffd728c _ZZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEiENK3$_0clEv + 156",
    "12  BreezSDKExample                     0x000000010ffd71e9 ___ZN8facebook5react15RCTNativeModule6invokeEjON5folly7dynamicEi_block_invoke + 25",
    "13  libdispatch.dylib                   0x0000000115fa8d18 _dispatch_call_block_and_release + 12",
    "14  libdispatch.dylib                   0x0000000115fa9f5b _dispatch_client_callout + 8",
    "15  libdispatch.dylib                   0x0000000115fb1d41 _dispatch_lane_serial_drain + 1064",
    "16  libdispatch.dylib                   0x0000000115fb2b1b _dispatch_lane_invoke + 441",
    "17  libdispatch.dylib                   0x0000000115fc0f13 _dispatch_workloop_worker_thread + 959",
    "18  libsystem_pthread.dylib             0x00007ff836172ce3 _pthread_wqthread + 326",
    "19  libsystem_pthread.dylib             0x00007ff836171c67 start_wqthread + 15"
  ]
}
```

Reply:
OK, I guess nothing can be done about that. This will work too!